From 59d579de4b833182723319b01b860c36d0e08772 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Thu, 1 Dec 2022 21:14:05 +0100
Subject: [PATCH 2/2] Fix a lot of Huawei EMUI9 denied under android 13 -
 update 2

- Add missing splash2 rules
- Add hwemerffu_service rules (CPU gouvernor)
- Add unknow rules
---
 sepolicy/huawei.te | 80 +++++++++++++++++++++++++++++++++++-----------
 1 file changed, 62 insertions(+), 18 deletions(-)

diff --git a/sepolicy/huawei.te b/sepolicy/huawei.te
index 6e15e7a..96324d8 100644
--- a/sepolicy/huawei.te
+++ b/sepolicy/huawei.te
@@ -45,27 +45,48 @@ type splash2_data_file, file_type;
 allow splash2_data_file self:filesystem { associate };
 allow init splash2_data_file:filesystem { mount relabelto relabelfrom };
 allow init splash2_data_file:dir { getattr mounton };
-allow vendor_init splash2_data_file:dir { open read setattr getattr };
-allow vendor_init splash2_data_file:file { open write create setattr getattr };
+allow vendor_init splash2_data_file:dir rw_dir_perms;
+allow vendor_init splash2_data_file:file rw_file_perms;
 allow vendor_init splash2_data_file:filesystem { getattr };
 allow installd splash2_data_file:filesystem { quotaget };
-allow rild splash2_data_file:dir { open read open };
-allow rild splash2_data_file:file { open read write getattr };
+allow rild splash2_data_file:dir rw_dir_perms;
+allow rild splash2_data_file:file rw_file_perms;
+allow kernel splash2_data_file:dir rw_dir_perms;
+allow kernel splash2_data_file:file rw_file_perms;
 
 
+# Fix CPU gouvernor (rules are in xml file)
+type hwemerffu_service, domain;
+allow hwemerffu_service proc:file r_file_perms;
 
-# Fix all SEdenied 
 
+# Fix all SEdenied 
 # allow vold modem_secure_file:filesystem { unmount };
-# allow hwemerffu_service proc:file { open read };
-# allow netutils_wrapper hinetmanager:fd { use };
-# allow hinetmanager self:capability { dac_override };
+
+type hinetmanager, fs_type, sysfs_type;
+allow netutils_wrapper hinetmanager:fd { use };
+allow netutils_wrapper hinetmanager:fifo_file { write };
+allow hinetmanager self:capability { dac_override };
+
+allow mediaprovider_app cache_file:dir r_dir_perms;
+
+
+type hal_usb_default, domain;
+allow hal_usb_default self:capability { dac_override };
+
+type sysfs_led, domain;
+allow init sysfs_led:file { setattr };
+
+
+#allow crash_dump bluetooth_data_file:file r_dir_perms;
+#allow crash_dump resourcecache_data_file:file { read };
+#allow crash_dump resourcecache_data_file:file { open };
 
 allow vold sys_block_mmcblk0:file { write };
 allow vold sysfs_zram:file { write };
 
-allow kernel device:dir { add_name write };
-allow kernel self:capability { mknod };
+allow kernel device:dir rw_dir_perms;
+allow kernel self:capability { mknod fsetid };
 allow kernel device:chr_file { create setattr };
 
 type teecd_data_file_system, data_file_type;
@@ -87,16 +108,18 @@ allow oeminfo_nvm block_device:blk_file { ioctl getattr open read write };
 allow gpsdaemon system_data_file:lnk_file { read };
 allow oeminfo_nvm device:chr_file { ioctl open read write };
 allow rild system_data_file:lnk_file { read };
-allow kernel system_data_root_file:dir { open read setattr };
-allow kernel system_data_root_file:file { open setattr getattr read append };
+allow kernel system_data_root_file:dir r_dir_perms;
+allow kernel system_data_root_file:file rw_file_perms;
 allow kernel unlabeled:file { read open };
 
+
 allow hi110x_daemon rootfs:dir { open read };
 allow hi110x_daemon self:fifo_file { ioctl };
 
+
 allow hal_camera_default system_data_file:lnk_file { read };
 allow surfaceflinger bootanim:dir { search };
-allow surfaceflinger bootanim:file { open read };
+allow surfaceflinger bootanim:file r_file_perms;
 allow hal_audio_default audioserver:fifo_file { write };
 allow installd self:capability { sys_ptrace };
 allow installd system_server:file { open read };
@@ -104,20 +127,31 @@ allow installd system_server:dir { search };
 
 allow installd teecd_data_file:filesystem { quotaget };
 allow system_server sysfs_zram:lnk_file { read };
-allow system_server vendor_file:file { getattr open read };
-allow kernel sysfs_devices_system_cpu:file { write };
-allow network_stack fs_bpf:file { read write };
+allow system_server vendor_file:file r_file_perms;
+allow kernel sysfs_devices_system_cpu:file rw_file_perms;
+allow network_stack fs_bpf:file rw_file_perms;
 
-# OTPERASE
+# OTPERASE = 0x4d19
 allowxperm oeminfo_nvm device:blk_file ioctl {
   0x4d19
 };
 
+# TIOCGWINSZ = 0x5413
+allowxperm hi110x_daemon self:fifo_file ioctl {
+  TIOCGWINSZ
+};
+
+# BLKROSET = 0x125d 0x127c
+allowxperm phhsu_daemon device:blk_file ioctl {
+  BLKROSET
+  BLKDISCARDZEROES
+};
+
+
 allow shell self:netlink_kobject_uevent_socket { create setopt getopt bind read };
 allow init self:netlink_kobject_uevent_socket { create setopt getopt bind read };
 
 
-
 # Authorize a lot of process to read default, config and system prop
 type hal_health_default, domain;
 type hal_camera_default, domain;
@@ -150,3 +184,13 @@ get_prop(gpsdaemon, default_prop)
 get_prop(oeminfo_nvm, default_prop)
 get_prop(atcmdserver, default_prop)
 
+type hal_drm_widevine, domain;
+type hisecd, domain;
+type storage_info, domain;
+
+get_prop(hal_drm_widevine, default_prop)
+get_prop(hisecd, default_prop)
+get_prop(mediaserver, vendor_default_prop)
+get_prop(storage_info, default_prop)
+
+
-- 
2.25.1

